src/irvisualizer/html_template_StmtToHTML.js

```diff
@@ -0,0 +1,151 @@
+            console.log(event.target);
+
+            currentResizer = event.target;
+            currentResizerIndex = resizeBars.indexOf(currentResizer);
+            var rect = currentResizer.getBoundingClientRect();
+            mousedown = true;
+            resizerInitCenter = event.x;
+            resizerPreviewX = event.x;
+            resizerPreview.style.left = resizerInitCenter + 'px';
+            resizerPreview.style.display = 'block';
+        }
+    });
+}
+
+document.addEventListener('mouseup', (event) => {
+    if (mousedown) {
+
+        currentResizer = null;
+        mousedown = false;
+        resizerPreview.style.display = 'none';
+
+        // Gather tab widths
+        var tabs = Array.from(document.querySelectorAll('div.tab'));
+        var widths = tabs.map(function(tab) {
+            return tab.getBoundingClientRect().width;
+        });
+
+
+        // Adjust tab widths
+        var diff = resizerPreviewX - resizerInitCenter;
+        console.log(diff);
+        var currentIndex = currentResizerIndex;
+        while (currentIndex >= 0 && tabs[currentIndex].classList.contains('collapsed-tab')) {
+            currentIndex--
+        }
+        widths[currentIndex] += diff;
+
+        currentIndex = currentResizerIndex + 1;
+        while (currentIndex < tabs.length && tabs[currentIndex].classList.contains('collapsed-tab')) {
+            currentIndex++;
+        }
+        widths[currentIndex] -= diff;
+
+        // Apply tab widths
+        tabs.forEach(function(tab, index) {
+            if (widths[index] >= 0) {
+                tab.style.width = widths[index] + 'px';
+            } else {
+                tab.style.width = '0px'; // or any other default value you want to set
+            }
+        });
+    }
+
+    document.body.classList.remove("no-select");
+});
+
+document.addEventListener('mousemove', (event) => {
+    if (mousedown) {
+        resizerPreview.style.left = event.x + 'px';
+        resizerPreviewX = event.x;
+    }
+});
+
+
+function collapse_tab(index) {
+    var tabs = document.getElementById("visualization-tabs");
+    var tab = tabs.firstElementChild.nextElementSibling;
+    for (var i = 0; i < index; ++i) {
+        tab = tab.nextElementSibling.nextElementSibling;
+    }
+
+    tab.classList.toggle('collapsed-tab');
+}
+
+function scrollToHostAsm(lno) {
+    var asmContent = document.getElementById("assemblyContent");
+    var line_height = window.getComputedStyle(asmContent).getPropertyValue("line-height");
+    line_height = parseInt(line_height, 10);
+    document.getElementById("host-assembly-tab").scrollTo({
+        behavior: "smooth",
+        top: lno * line_height
+    });
+}
+
+function scrollToDeviceCode(lno) {
+    var device_code_tab = document.getElementById("device-code-tab");
+    const lineSpans = device_code_tab.querySelectorAll('span.line');
+    line = lineSpans[lno - 1]
+    line.scrollIntoView({
+        behavior: "smooth",
```

Comment:
My linter catches some errors; all are minute. I am attaching the log here for your reference.

```javascript
/home/antony/Projects/halide/src/irvisualizer/html_template_StmtToHTML.js
   33:10  error  'scrollToCode' is defined but never used        no-unused-vars
   36:5   error  'makeCodeVisible' is not defined                no-undef
   57:13  error  Unexpected console statement                    no-console
   61:17  error  'rect' is defined but never used                no-unused-vars
   71:39  error  'event' is defined but never used               no-unused-vars
   85:9   error  Unexpected console statement                    no-console
  119:10  error  'collapse_tab' is defined but never used        no-unused-vars
  129:10  error  'scrollToHostAsm' is defined but never used     no-unused-vars
  137:10  error  'scrollToDeviceCode' is defined but never used  no-unused-vars
  140:5   error  'line' is not defined                           no-undef
  141:5   error  'line' is not defined                           no-undef
  142:28  error  Unexpected trailing comma                       comma-dangle
```

Reply:
This is the content of my `.eslint.js` config:

```javascript
module.exports = {
  env: {
    es6: true,
    node: true,
    browser: true
  },
  extends: [
    'eslint:recommended',
  ],
  parserOptions: {
    // Enable JSX support
    ecmaFeatures: { jsx: true },
    ecmaVersion: 2018,
    sourceType: 'module'
  },
  //parser: 'babel-eslint',
  rules: {
    // Add any additional rules here
    'react/prop-types': 0
  }
}
```

Reply:
Thank you for these! I will look into it. Is there ways to suppress some of these? Some of these "unused functions" are of course not unused, as the HTML is sprinkled with `onclick="scrollToHostAsm()"` for example.

Reply:
Yeah, I found that this should work:

```js
function scrollToDeviceCode(lno) {  // eslint-disable-line no-unused-vars
```

Reply:
> I haven't looked at running eslint myself, but if you'd like to give it another go, feel free to!

For sure. I am linting the `StmtToHTML.js` only. 

```javascript
halide/src/irvisualizer/html_template_StmtToHTML.js
   34:5   error  'makeCodeVisible' is not defined   no-undef
   69:39  error  'event' is defined but never used  no-unused-vars
  146:51  error  'device_code_Tab' is not defined   no-undef
```

@mcourteaux Ping me again if you want a sanity check for legacy web browsers:

![image](https://github.com/halide/Halide/assets/12185630/d2ad995d-22ab-4208-8ec1-112bd94a0e1f)

Reply:
I don't know how to interpret the screenshot of this legacy browser check. There are a bunch of numbers, but I don't know what they all indicate. Please keep in mind that this Halide pipeline visualizer HTML is a tool for developers who work on a desktop. The browsers we care about are all desktop based: Firefox, Chrome, Edge, Safari, Opera, would probably cover most of them.